src/backend/joanie/core/serializers.py

```diff
@@ -1,28 +1,183 @@
 """Serializers for api."""
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+
 from rest_framework import serializers
 
-from joanie.core import models
+from joanie.core import enums, models
+
+
+class CertificationDefinitionSerializer(serializers.ModelSerializer):
+    """
+    Serialize information about a certificate definition
+    """
+
+    description = serializers.CharField(allow_blank=True, allow_null=True)
+
+    class Meta:
+        model = models.CertificateDefinition
+        fields = ["description", "name", "title"]
+
+
+class OrganizationSerializer(serializers.ModelSerializer):
+    """
+    Serialize all non-sensitive information about an organization
+    """
+
+    class Meta:
+        model = models.Organization
+        fields = ["code", "title"]
+
+
+class TargetCourseSerializer(serializers.ModelSerializer):
+    """
+    Serialize all information about a target course.
+    """
+
+    course_runs = serializers.SerializerMethodField(read_only=True)
+    organization = OrganizationSerializer(read_only=True)
+    position = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = models.Course
+        fields = [
+            "code",
+            "course_runs",
+            "organization",
+            "position",
+            "title",
+        ]
+
+    def get_position(self, target_course):
+        """
+        Retrieve the position of the course related to its product_relation
+        """
+        product = self.context.get("product", None)
+        return target_course.product_relations.get(product=product).position
+
+    def get_course_runs(self, target_course):
+        """
+        For a given course_run, retrieve enrollment
+        related to the authenticated user and the context product.
+        If one exist, replace the course run object by the enrollment
+        """
+        user = self.context.get("user", None)
+        product = self.context.get("product", None)
+        course_runs = CourseRunSerializer(
+            target_course.course_runs.all().order_by("start"), many=True
+        ).data
+
+        if isinstance(user, get_user_model()):
+            for index, course_run in enumerate(course_runs):
+                try:
+                    enrollment = models.Enrollment.objects.get(
+                        course_run__pk=course_run["id"],
+                        user=user,
+                        order__product=product,
+                    )
+                    course_runs[index] = CourseRunEnrollmentSerializer(enrollment).data
```

Comment:
You can't use index from enumerate because, for some products, you will go through the ObjectDoesNotExist exception and the index will still be incremented? Try adding a test to surface the problem.

Reply:
I thought of it more and am not confortable with this field holding different types of objects depending of the situation. I think we should have separate "enrollments" field for that. WDYT?

Reply:
Yes ! That's the problem I was originally trying to solve that I noticed this. It's easier to understand as well.

Reply:
Ok now I simply retrieve enrollments related to the order.